auth/service/api/v1/profile_filter.go

```diff
@@ -0,0 +1,227 @@
+package v1
+
+import (
+	"net/url"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/tidepool-org/platform/permission"
+	"github.com/tidepool-org/platform/pointer"
+	userLib "github.com/tidepool-org/platform/user"
+)
+
+var (
+	ANY   = []string{"any"}
+	NONE  = []string{"none"}
+	TRUES = []string{"true", "yes", "y", "1"}
+)
+
+type parsedQueryPermissions []string
+
+type usersProfileFilter struct {
+	TrustorPermissions parsedQueryPermissions
+	TrusteePermissions parsedQueryPermissions
+	Email              *regexp.Regexp
+	EmailVerified      *bool
+	TermsAccepted      *regexp.Regexp
+	Name               *regexp.Regexp
+	Birthday           *regexp.Regexp
+	DiagnosisDate      *regexp.Regexp
+}
+
+// parsePermissions replicates the functionality of the seagull node.js
+// parsePermissions function which parses a query string value to a permissions
+// slice
+func parsePermissions(queryValuesOfKey string) parsedQueryPermissions {
+	queryValuesOfKey = strings.TrimSpace(queryValuesOfKey)
+	if queryValuesOfKey == "" {
+		return nil
+	}
+	var vals []string
+	for _, val := range strings.Split(queryValuesOfKey, ",") {
+		val = strings.TrimSpace(val)
+		// Remove falsey values that are strings
+		if val == "0" || val == "false" || val == "" {
+			continue
+		}
+		vals = append(vals, val)
+	}
+	if slices.Compare(vals, ANY) == 0 {
+		return slices.Clone(ANY)
+	}
+	if slices.Compare(vals, NONE) == 0 {
+		return slices.Clone(NONE)
+	}
+	for _, val := range vals {
+		nonEmpty := val != ""
+		if !nonEmpty {
+			return vals
+		}
+	}
+	return nil
+}
+
+// This is the logic I'm most unsure about.
+// It's seems to be saying if a parsed query contains either of ANY or NONE that it is not a valid permissions query.
+// Yet later in arePermissionsValid it allows those values.
+func arePermissionsValid(perms parsedQueryPermissions) bool {
+	if len(perms) > 1 {
+		union := append(ANY, NONE...)
+		for _, perm := range perms {
+			// quadratic time complexity but very few elements so don't care
+			if slices.Contains(union, perm) {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func arePermissionsSatisfied(queryPermissions parsedQueryPermissions, userPermissions permission.Permission) bool {
+	if slices.Compare(queryPermissions, ANY) == 0 {
+		nonEmpty := len(userPermissions) > 0
+		return nonEmpty
+	}
+	if slices.Compare(queryPermissions, NONE) == 0 {
+		empty := len(userPermissions) == 0
+		return empty
+	}
+	// Todo: really test this part
+	for _, queryPerm := range queryPermissions {
+		if _, ok := userPermissions[queryPerm]; !ok {
+			return false
+		}
+	}
+	return true
+}
+
+func stringToBoolean(value string) bool {
+	value = strings.TrimSpace(strings.ToLower(value))
+	return slices.Contains(TRUES, value)
+}
+
+func parseUsersQuery(query url.Values) *usersProfileFilter {
```

Comment:
My impressions is that this query is not used at all. Can we check blip and uploader?

Reply:
On checking blip and uploader it seems they are indeed no query string parameters used.

[blip](https://github.com/tidepool-org/blip/blob/develop/app/core/api.js#L426)
[platform-client](https://github.com/tidepool-org/platform-client/blob/develop/index.js#L334)
[uploader](https://github.com/tidepool-org/uploader/blob/master/lib/core/api.js#L274)
[uploader](https://github.com/tidepool-org/uploader/blob/master/lib/core/api.js#L315)

Can you confirm that no query string parameters are passed for `getAssociatedUsersDetails` @krystophv @gniezen @jh-bate ?